python/cudf/cudf/tests/test_parquet.py

```diff
@@ -316,6 +319,102 @@ def num_row_groups(rows, group_size):
         assert a == b
 
 
+def test_parquet_read_filtered(tmpdir):
+    # Generate data
+    fname = tmpdir.join("filtered.parquet")
+    dg.generate(
+        fname,
+        dg.Parameters(
+            num_rows=2048,
+            column_parameters=[
+                dg.ColumnParameters(
+                    cardinality=100,
+                    null_frequency=0.05,
+                    ty=lambda g: g.address.country,
+                    is_sorted=False,
+                ),
+                dg.ColumnParameters(40, 0.2, lambda g: g.person.age, True,),
+            ],
+            seed=0,
+        ),
+    )
+
+    # Get dataframes to compare
+    df = cudf.read_parquet(fname)
+    df_filtered = cudf.read_parquet(fname, filters=[("1", ">", 60)])
+
+    assert cudf.io.read_parquet_metadata(fname)[1] == 2048 / 64
+    assert len(df_filtered) < len(df)
```

Comment:
Instead of just asserting this length, could we read the file via pyarrow, get the length that way and assert the length is the same for cudf?

Reply:
I don't think so - PyArrow does have a `read_table` function that accepts `filters`. But this is used to select and read row groups into memory _and_ apply filters in memory. The only way to use PyArrow to _only_ do row-group-level filtering is to re-use my row group selection code [1] and then passing that set of row groups on to something that isn't cuDF to read. But that would be pointless and effectively just testing whether or not cuDF can read a given set of row groups, which is already tested.

[1] https://github.com/rapidsai/cudf/pull/5843/files#diff-deac873508aaa12ca2e7c0a2c9035230R216-R221

Reply:
Got it, thanks for the explanation. Could you add a comment along those lines to the test?